cookbooks/aws-parallelcluster-slurm/resources/s3_object/s3_object.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at http://aws.amazon.com/apache2.0/
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+resource_name :s3_object
+provides :s3_object
+unified_mode true
+
+# Resource to retrieve an S3 Object either using the S3 or HTTPS URI format
+property :url, required: true,
+         description: 'Source URI of the remote file'
+property :destination, required: true,
+         description: 'destination path where to store the file'
+
+default_action :get
+
+action :get do
+  if !new_resource.url.empty? && !new_resource.destination.empty?
+    source_url = new_resource.url
+    local_path = new_resource.destination
+    Chef::Log.debug("Retrieving S3 Object from #{source_url} to #{local_path} using S3 protocol")
+
+    # if running a test skip credential check
+    no_sign_request = kitchen_test? ? "--no-sign-request" : ""
```

Comment:
This line should probably go inside the if clause (it should be relevant only for the S3 protocol, shouldn't it?).

Reply:
moved